lib/saddler/reporter/github/git_repository.rb

```diff
@@ -0,0 +1,49 @@
+module Saddler
+  module Reporter
+    module Github
+      class GitRepository
+        def initialize(path)
+          @git = Git.open(path)
+        end
+
+        def slug
+          name = /[[:alnum:]_\-\.]*/
+          repo = /[[:alnum:]_\-\.]*/
+          regex_slug = %r{#{name}/#{repo}}
+          regex = /.*?#{Regexp.quote(github_domain)}.*?(?<slug>#{regex_slug})/
+          target = remote_urls.map do |url|
+            match = url.match(regex)
+            match[:slug] if match
+          end.compact.first
+          strip_git_extension(target)
+        end
+
+        def branch
+        end
+
+        def remote_urls
+          @git
+            .branches
+            .remote
+            .map { |branch| branch.remote.url }
+        end
+
+        def strip_git_extension(name)
+          /\A(?<identity>.*?)(?:\.git)?\z/ =~ name
+          identity
+        end
+
+        def github_domain
```

Comment:
I think this method return invalid value if set endpoint. e.g. https:://your.example.com/api/v3